fire/SFMainMod.F90

```diff
@@ -204,7 +189,78 @@ subroutine UpdateFuelCharacteristics(currentSite)
+    !
+    
+    use FatesInterfaceTypesMod, only : hlm_spitfire_mode
+    use EDParamsMod,            only : cg_strikes
+    use EDParamsMod,            only : ED_val_nignitions
+    use SFParamsMod,            only : SF_val_fdi_alpha
+    use FatesConstantsMod,      only : years_per_day
+
+    ! ARGUMENTS:
+    type(ed_site_type), intent(inout), target :: currentSite ! site object
+    type(bc_in_type),   intent(in)            :: bc_in       ! BC in object
+    
+    ! LOCALS:
+    type(fates_patch_type), pointer :: currentPatch            ! patch object
+    real(r8)                        :: cloud_to_ground_strikes ! fraction of cloud-to-ground strikes [0-1]
+    real(r8)                        :: anthro_ignitions        ! anthropogenic ignitions [count/km2/day]
+    integer                         :: iofp                    ! patch index
+    
+    ! CONSTANTS:
+    real(r8), parameter :: alpha = 0.0035_r8  ! potential human ignition counts (alpha in Li et al. 2012) (#/person/month)
+
+    ! initialize site parameters to zero
+    currentSite%NF_successful = 0.0_r8
+
+    ! Equation 7 from Venevsky et al GCB 2002 (modification of equation 8 in Thonicke et al. 2010) 
+    ! FDI 0.1 = low, 0.3 moderate, 0.75 high, and 1 = extreme ignition potential for alpha 0.000337
+    if (hlm_spitfire_mode == hlm_sf_successful_ignitions_def) then
+      ! READING "SUCCESSFUL IGNITION" DATA
+      ! force ignition potential to be extreme
+      ! cloud_to_ground_strikes = 1 means using 100% of incoming observed ignitions
+      currentSite%FDI = 1.0_r8  
+      cloud_to_ground_strikes = 1.0_r8   
+    else  
+      ! USING LIGHTNING STRIKE DATA
+      currentSite%FDI  = 1.0_r8 - exp(-SF_val_fdi_alpha*currentSite%fireWeather%fire_weather_index)
+      cloud_to_ground_strikes = cg_strikes
+    end if
+
+    ! if the oldest patch is a bareground patch (i.e. nocomp mode is on) use the first vegetated patch
+    ! for the iofp index (i.e. the next younger patch)
+    currentPatch => currentSite%oldest_patch
+    if(currentPatch%nocomp_pft_label .eq. nocomp_bareground)then
+      currentPatch => currentPatch%younger
+    endif
+    iofp = currentPatch%patchno
+
+    ! NF = number of lighting strikes per day per km2 scaled by cloud to ground strikes
+    if (hlm_spitfire_mode == hlm_sf_scalar_lightning_def) then
+      currentSite%NF = ED_val_nignitions*years_per_day*cloud_to_ground_strikes
+    else    
+      ! use external daily lightning ignition data
+      currentSite%NF = bc_in%lightning24(iofp)*cloud_to_ground_strikes
+    end if
+
+    ! calculate anthropogenic ignitions according to Li et al. (2012)
+    ! add to ignitions by lightning
+    if (hlm_spitfire_mode == hlm_sf_anthro_ignitions_def) then
+      ! anthropogenic ignitions (count/km2/day)
+      !           =  (ignitions/person/month)*6.8*population_density**0.43/approximate days per month
+      anthro_ignitions = alpha*6.8_r8*bc_in%pop_density(iofp)**0.43_r8/30.0_r8
```

Comment:
`30.0_r8` here is a magic number.

Reply:
I think it's approximate days per month? I can make it a parameter

Reply:
Yeah, that's what I gathered from the comment. But the thing about magic numbers is that they can eventually get separated from the explanatory comments, so it's best that they're named variables.